ntp-daemon/src/system.rs

```diff
@@ -552,6 +624,48 @@ impl Spawner {
                 address: config.addr,
             },
             address: addr,
+            nts: None,
+        };
+
+        if let Err(send_error) = sender.send(spawn_task).await {
+            tracing::error!(?send_error, "Receive half got disconnected");
+        }
+    }
+
+    async fn spawn_nts(
+        ke: KeyExchangeResult,
+        extra_certificates: Arc<[Certificate]>,
+        sender: Sender<SpawnTask>,
+    ) {
+        let addr = loop {
+            let address = (ke.remote.as_str(), ke.port);
+            match tokio::net::lookup_host(address).await {
+                Ok(mut addresses) => match addresses.next() {
+                    None => {
+                        warn!("Could not resolve peer address, retrying");
+                        tokio::time::sleep(NETWORK_WAIT_PERIOD).await
+                    }
+                    Some(first) => {
+                        break first;
+                    }
+                },
+                Err(e) => {
+                    warn!(error = ?e, "error while resolving peer address, retrying");
+                    tokio::time::sleep(NETWORK_WAIT_PERIOD).await
+                }
+            }
+        };
+
+        // we assume that when KE succeeds, it gives back a structurally valid remote and port.
+        let address = NormalizedAddress::new_unchecked(&ke.remote, ke.port);
```

Comment:
Given that the remote can be provided by the server, blindly assuming here is maybe not the best option. What are the consequences if the remote part is not valid?